lib/mapper.js

```diff
@@ -54,11 +56,15 @@ exports.track = function(track){
   }];
 
   // Formulate properties to required specs
-  // TODO handle nested objects and arrays by flattening
   events[0].properties = foldl(function(results, value, key){
-    if (value) results[key] = [value.toString()]; // they do not accept any other data type
+    if (value) {
+      if (!is.array(value)) value = [value];
+      results[key] = value.map(function(prop){
+        return prop.toString();
+      });
+    }
     return results;
-  }, {}, track.properties());
+  }, {}, trample(track.properties()));
```

Comment:
It would probably be easier to read if you split the functions on separate lines.

```js
var trampledObject = trample(...);
events[0].properties = fold1(..., {}, trampledObject);
```